docs/external-node.md

```diff
@@ -0,0 +1,569 @@
+  permission is moved to `antrea-controller`
+- for more details please refer to [vm-agent-rbac.yml](/build/yamls/externalnode/vm-agent-rbac.yml)
+
+`antrea-agent` reports its status by updating the `antreaagentinfo` resource
+which is created with the same name as the `ExternalNode`. `antrea-controller`
+creates an `antreaagentinfo` resource for each new `ExternalNode`, and then
+`antrea-agent` updates it every minute with its latest status. `antreaagentinfo`
+is deleted by `antrea-controller` when the `ExternalNode` is deleted.
+
+## Apply Antrea NetworkPolicy to ExternalNode
+
+### Antrea NetworkPolicy configuration
+
+An Antrea NetworkPolicy is applied to an `ExternalNode` by providing an
+`externalEntitySelector` in the `appliedTo` field. **The `ExternalEntity`
+resource is automatically created for each interface of an `ExternalNode`**,
+which are used by `antrea-controller` to calculate the NetworkPolicies, and
+`antrea-agent` running on the `ExternalNode` receives the internal Antrea
+NetworkPolicy objects.
+
+Following types of (from/to) network peers are supported in an Antrea
+NetworkPolicy applied to an external Node:
+
+- ExternalEntities selected by an `externalEntitySelector`
+- An `ipBlock`
+- A FQDN address in an egress rule
+
+Following actions are supported in an Antrea NetworkPolicy applied to an
+external Node:
+
+- Allow
+- Drop
+- Reject
+
+Below is an example of applying an Antrea NetworkPolicy to the external Nodes
+labeled with `role=db` to reject SSH connections from IP "172.16.100.5" or from
+other external Nodes labeled with `role=front`:
+
+```yaml
+kind: NetworkPolicy
+metadata:
+  name: anp1
+  namespace: vm-ns
+spec:
+  priority: 9000.0
+  appliedTo:
+    - externalEntitySelector:
+        matchLabels:
+          role: db
+  ingress:
+    - action: Reject
+      ports:
+        - protocol: TCP
+          port: 22
+      from:
+        - externalEntitySelector:
+            matchLabels:
+              role: front
+        - ipBlock:
+            cidr: 172.16.100.5/32
+```
+
+### Bypass Antrea NetworkPolicy
+
+In some cases, the special IP traffic is expected to bypass the Antrea
+NetworkPolicy rules on an external Node, e.g.,the SSH connection from a special
+host to the external Node. These rules can be set with the item
+`policyBypassRules` in the agent  configuration. Following is a configuration
+example,
+
+```yaml
+policyBypassRules:
+  - direction: ingress
+    protocol: tcp
+    cidr: 1.1.1.1/32
+    port: 22
+```
+
+The `direction` in one policyBypassRule supports `ingress` or `egress`. And
+the supported protocols include, `tcp`,`udp`, `icmp` and `ip`. The `cidr`
+gives the peer address, it is the destination in an `egress` rule, and the
+source in an `ingress` rule. For `tcp` and `udp` protocols, the `port` is
+required to describe the destination port in the packet.
+
+The policyBypassRules are processed before Antrea NetworkPolicy rules when
```

Comment:
I feel we should remove this sentence or move to a proper place in "OpenFlow pipeline"

Reply:
removed this sentence.